app/notification_push.go

```diff
@@ -239,6 +206,32 @@ func (a *App) ClearPushNotification(currentSessionId, userId, channelId string)
 	}
 }
 
+func (a *App) UpdateMobileAppBadgeSync(userId string) *model.AppError {
+	msg := model.PushNotification{
+		Type:             model.PUSH_TYPE_UPDATE_BADGE,
+		Version:          model.PUSH_MESSAGE_V2,
+		Sound:            "none",
+		ContentAvailable: 1,
+	}
+
+	if unreadCount, err := a.Srv.Store.User().GetUnreadCount(userId); err != nil {
+		msg.Badge = 0
```

Comment:
if we can't get the unread count, shouldn't we just fail with an AppError instead of pushing a 0?

Reply:
Yeah, I'll look at doing that. I need to look at the error handling around these functions as well since I've been copying from the other functions in this file, and it seems like the returned error values are just ignored as well